tools/restatectl/src/connection.rs

```diff
@@ -184,25 +196,48 @@ impl ConnectionInfo {
                 .accept_compressed(CompressionEncoding::Gzip)
                 .send_compressed(CompressionEncoding::Gzip);
 
-            let response = match client.get_ident(()).await {
+            let ident_response = match client.get_ident(()).await {
                 Ok(response) => response.into_inner(),
                 Err(status) => {
                     errors.error(address.clone(), status);
                     self.dead_nodes.write().unwrap().insert(address.clone());
                     continue;
                 }
             };
-            ident_responses.insert(address.clone(), response.clone());
+
+            update_required(&ident_response);
+
+            ident_responses.insert(address.clone(), ident_response.clone());
+
+            if Self::metadata_proxy_supported(&ident_response) {
+                // short circuit to reading directly from the proxy
+                match self.metadata_proxy_get(channel.clone(), kind).await {
+                    Err(status) => {
+                        errors.error(address.clone(), status);
+                        continue;
+                    }
+                    Ok(Some(value)) => return Ok(value),
+                    Ok(None) => {
+                        // we don't need to try again if value does not exist
+                        return Err(ConnectionInfoError::MetadataValueNotAvailable {
+                            contacted_nodes: ident_responses,
+                        });
+                    }
+                }
+            }
+
+            // if metadata proxy is not supported, we can
+            // continue with the majority read approach
```

Comment:
I was thinking - what is the value of doing a majority read over requesting a consistent / linearizable read by setting the sync flag? Isn't that going to be more efficient _and_ more likely to give you a correct result? Using the reader majority as a proxy may not even be "correct" depending on the metadata store implementation in use.

Reply:
It was done like that in case one (or multiple) nodes are partitioned from the rest of the cluster and or can't reach the metadata-server

Reply:
I think we can lean on the metadata client implementations to take care of this, if necessary! If we're doing a "sync" read which forces a reload, the underlying metadata store must be able to give us a linearizable read - or correctly fail with an error, if it's partitioned and can't guarantee it. We can lean on this property? And if you're in YOLO mode, then fine, just use the first cached response we get.